homeassistant/components/sensor/flunearyou.py

```diff
@@ -0,0 +1,212 @@
+https://home-assistant.io/components/sensor.flunearyou/
+"""
+import logging
+from datetime import timedelta
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (
+    ATTR_ATTRIBUTION, ATTR_STATE, CONF_LATITUDE, CONF_MONITORED_CONDITIONS,
+    CONF_LONGITUDE)
+from homeassistant.helpers import aiohttp_client
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+
+REQUIREMENTS = ['pyflunearyou==0.0.2']
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_CITY = 'city'
+ATTR_REPORTED_DATE = 'reported_date'
+ATTR_REPORTED_LATITUDE = 'reported_latitude'
+ATTR_REPORTED_LONGITUDE = 'reported_longitude'
+ATTR_ZIP_CODE = 'zip_code'
+
+DEFAULT_ATTRIBUTION = 'Data provided by Flu Near You'
+DEFAULT_SCAN_INTERVAL = timedelta(minutes=30)
+
+CATEGORY_CDC_REPORT = 'cdc_report'
+CATEGORY_USER_REPORT = 'user_report'
+
+TYPE_CDC_LEVEL = 'level'
+TYPE_CDC_LEVEL2 = 'level2'
+TYPE_USER_CHICK = 'chick'
+TYPE_USER_DENGUE = 'dengue'
+TYPE_USER_FLU = 'flu'
+TYPE_USER_LEPTO = 'lepto'
+TYPE_USER_NO_NONE = 'none'
+TYPE_USER_SYMPTOMS = 'symptoms'
+TYPE_USER_TOTAL = 'total'
+
+SENSORS = {
+    CATEGORY_CDC_REPORT: [
+        (TYPE_CDC_LEVEL, 'CDC Level', 'mdi:biohazard', None),
+        (TYPE_CDC_LEVEL2, 'CDC Level 2', 'mdi:biohazard', None),
+    ],
+    CATEGORY_USER_REPORT: [
+        (TYPE_USER_CHICK, 'Avian Flu Symptoms', 'mdi:alert', 'reports'),
+        (TYPE_USER_DENGUE, 'Dengue Fever Symptoms', 'mdi:alert', 'reports'),
+        (TYPE_USER_FLU, 'Flu Symptoms', 'mdi:alert', 'reports'),
+        (TYPE_USER_LEPTO, 'Leptospirosis Symptoms', 'mdi:alert', 'reports'),
+        (TYPE_USER_NO_NONE, 'No Symptoms', 'mdi:alert', 'reports'),
+        (TYPE_USER_SYMPTOMS, 'Flu-like Symptoms', 'mdi:alert', 'reports'),
+        (TYPE_USER_TOTAL, 'Total Symptoms', 'mdi:alert', 'reports'),
+    ]
+}
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_LATITUDE):
+        cv.latitude,
+    vol.Optional(CONF_LONGITUDE):
+        cv.longitude,
+    vol.Required(CONF_MONITORED_CONDITIONS, default=list(SENSORS)):
+        vol.All(cv.ensure_list, [vol.In(SENSORS)])
+})
+
+
+async def async_setup_platform(
+        hass, config, async_add_entities, discovery_info=None):
+    """Configure the platform and add the sensors."""
+    from pyflunearyou import create_client
+    from pyflunearyou.errors import FluNearYouError
+
+    websession = aiohttp_client.async_get_clientsession(hass)
+
+    latitude = config.get(CONF_LATITUDE, hass.config.latitude)
+    longitude = config.get(CONF_LONGITUDE, hass.config.longitude)
+    identifier = '{0},{1}'.format(latitude, longitude)
+
+    try:
+        client = await create_client(latitude, longitude, websession)
+    except FluNearYouError as err:
+        _LOGGER.error('There was an error while setting up: %s', err)
+        return False
```

Comment:
If you expect that it might work later (like when the network is back), you can raise `PlatformNotReady`.

Reply:
:point_up: or just `return`. Nothing is checking this return value.